### PR TITLE
fix(edge): add image parsing for kube image as docker for private registries EE-3439

### PR DESCRIPTION
**Pull request by prabhat83**

### CODE REVIEW

File: edge/yaml/yaml.go [hunk 1]
```diff
@@ -34,7 +35,11 @@ func NewYAML(fileContent string, credentials []agent.RegistryCredentials) *yaml
 func (y *yaml) getRegistryCredentialsByImageURL(imageURL string) []agent.RegistryCredentials {
 	credentials := []agent.RegistryCredentials{}
 	for _, r := range y.registryCredentials {
-		if strings.Contains(imageURL, r.ServerURL) {
+		domain, _, err := getRegistryAndPath(imageURL)
```
Comment (dakota-portainer) [inline edge/yaml/yaml.go]: Any reason why this returns a `path` if we just throw it out and don't use this function anywhere else? Are you thinking you may want the path in the future?

**Review by dakota-portainer — APPROVED** [review]
Quick question. Otherwise LGTM.

File: edge/yaml/yaml.go [hunk 2]
```diff
@@ -34,7 +35,11 @@ func NewYAML(fileContent string, credentials []agent.RegistryCredentials) *yaml
...
+		if err != nil {
+			return nil
+		}
+		if strings.Contains(domain, r.ServerURL) || strings.Contains(r.ServerURL, domain) {
```
Comment (hookenz) [inline edge/yaml/yaml.go]: Might be good to explain this logic with a comment.   It basically sanitises the registry imageUrl.

e.g. 
imageUrl = prabhat/nginx   when passed through getRegistryAndPath will return "docker.io" for the domain.
While  imageUrl=quay.io/prabhat/nginx will return "quay.io" for the domain

Second check, we really just need to ensure the domain is in the serverUrl.

**Review by hookenz — APPROVED** [review]
Just a minor thing, but otherwise LGTM and approved ahead of time.

File: edge/yaml/yaml.go (hunk 1 above)
Comment (prabhat83) [reply edge/yaml/yaml.go]: you are right. I copied this function from Matt's work in docker and used it without thinking I guess. Good spotting.

**Pull request merged**